packages/server/graphql/mutations/messageAllSlackUsers.ts

```diff
@@ -0,0 +1,56 @@
+import {GraphQLNonNull, GraphQLString} from 'graphql'
+import getRethink from '../../database/rethinkDriver'
+import {requireSU} from '../../utils/authorization'
+import MessageAllSlackUsersPayload from '../types/MessageAllSlackUsersPayload'
+import {GQLContext} from '../graphql'
+import SlackServerManager from '../../utils/SlackServerManager'
+import standardError from '../../utils/standardError'
+
+interface MessageSlackUserError {
+  userId: string
+  error: string
+}
+
+const messageAllSlackUsers = {
+  type: GraphQLNonNull(MessageAllSlackUsersPayload),
+  description: 'Send a message to all authorised Slack users',
+  args: {
+    message: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: 'The slack message that will be sent to all Slack users'
+    }
+  },
+  resolve: async (_source, {message}, {authToken}: GQLContext) => {
+    const r = await getRethink()
+
+    //AUTH
+    requireSU(authToken)
```

Comment:
On the private schema, are all users with access also super users? If so, and if there are no plans to change that, I can remove this

Reply:
today that's true, barring any bugs. for things this important, it's a good idea to have every mutation be responsible for its own perms so when we do get hacked, it is harder for them to cause havoc